paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/EventClassProvider.java

```diff
@@ -3,15 +3,25 @@
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
 
+import java.util.Arrays;
+
 import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 import de.zalando.paradox.nakadi.consumer.core.domain.NakadiEventBatch;
+import de.zalando.paradox.nakadi.consumer.core.exceptions.UnrecoverableException;
 
 public interface EventClassProvider<T> {
-    Class<T> getEventClass();
 
-    static JavaType getJavaType(final EventClassProvider delegate, final ObjectMapper jsonMapper) {
+    default Class<T> getEventClass() {
+        return Arrays.stream(getClass().getGenericInterfaces()).filter(genericInterface ->
+                             genericInterface instanceof ParameterizedType).findFirst().map(genericInterface ->
+                             ((ParameterizedType) genericInterface).getActualTypeArguments()).filter(genericTypes ->
+                             genericTypes.length > 0).map(genericTypes -> (Class<T>) genericTypes[0]).orElseThrow(
```

Comment:
.findAny does not work there. It it an array and I should take one of the elements.